lib/features/pokemon_overview/pokemon_overview_page.dart

```diff
@@ -20,14 +21,18 @@ class PokemonOverviewPage extends StatelessWidget {
         centerTitle: true,
         backgroundColor: primaryColor,
       ),
-      body: GridView.builder(
-        padding: const EdgeInsets.all(10.0),
-        itemCount: pokemons.length,
-        gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(crossAxisCount: pokemonColumnCount),
-        itemBuilder: (_, index) {
-          final pokemon = pokemons[index];
-          return PokemonCard(pokemon: pokemon);
-        },
+      body: pokemons.when(
+        (data) => GridView.builder(
+          padding: const EdgeInsets.all(10.0),
+          itemCount: data.length,
+          gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(crossAxisCount: pokemonColumnCount),
+          itemBuilder: (_, index) {
+            final pokemon = data[index];
+            return PokemonCard(pokemon: pokemon);
+          },
+        ),
+        loading: () => const Center(child: CircularProgressIndicator()),
+        error: (errorMessage) => AlertDialog(title: Text('$errorMessage')),
```

Comment:
this seems not the proper way of calling the alert dialog 🙂 

https://api.flutter.dev/flutter/material/AlertDialog-class.html

Reply:
Hi @shawn-ffufm. I just want to ask because the error properties is expecting a widget and If i wrap the AlertDialog into a showDialog then it's giving me error message.

![image](https://user-images.githubusercontent.com/106600653/227858665-b9294748-cba0-4f98-8d2b-6ace5de06d29.png)

Reply:
```
error: (errorMessage) {
   _show....
}
```

Reply:
how about like that? 🙂